scripts/versioner.py

```diff
@@ -46,24 +46,9 @@ def undefined(arg, *args, **kwargs):
 JINJA_CONFIG = {'pin_compatible'        : undefined,
                 'pin_subpackage'        : undefined,
                 'compiler'              : undefined,
-                'base_mpich'            : undefined('mpich'),
-                'base_mpicc'            : undefined('mpicc'),
-                'base_mpicxx'           : undefined('mpicxx'),
-                'base_mpifort'          : undefined('mpifort'),
+                'mpi'                   : undefined('mpich|openmpi'),
```

Comment:
Only one question on this- no need to check `mpicc, mpicxx, mpifort` etc. here since we're checking against the variants? 

Otherwise nice simplification in `moose-mpi` and here!

Reply:
This is for both assigning a value to variables encountered by jinja templating, and for cosmetic reasons. The `mpi` key/value is actually not valid of course, but it allows us to print out something like this:

```bash
$ cd moose/scripts/
$ ./versioner.py mpi --yaml | grep install
  install: moose-mpi=2024.04.25=mpich|openmpi
$ 
```
I realize some people are going to literally copy/paste that entire line... But I am not sure how else to inform a person that they have a choice in the matter. I am totally open to suggestions!

Reply:
Spaces work here:
```bash
$ ./versioner.py mpi --yaml | grep install
  install: moose-mpi=2024.04.25=mpich or openmpi
```

Reply:
Oh, I have a better idea. I can read our `packages_config.yml` file to determine our "default" MPI wrapper of choice.

Reply:
Thanks for the clarification of what that bit of code is actually doing. Reading from `packages_config.yml` is fine to indicate a default I think. Though it would be nice to show that openmpi is an option for those who go far enough to run that command. 

Perhaps something like:

```
$ ./versioner.py mpi --yaml | grep install
  install: moose-mpi=2024.04.25={mpich || openmpi}
```

This would indicate to a developer the `OR`-ness of this choice to me in logic that we're used to seeing.